tf_quant_finance/black_scholes/vanilla_prices_test.py

```diff
@@ -334,6 +334,144 @@ def test_binary_vanilla_consistency_exact(self):
         tf.where(is_call_options, -implied_binary_price, implied_binary_price))
     self.assertArrayNear(implied_binary_price, actual_binary_price, 1e-10)
 
+  def test_barrier_option_1d_puo(self):
+    """Function tests barrier option pricing for scalar input"""
+    spots = 100.0
+    rebates = 3.0
+    expiries = 0.5
+    discount_rates = 0.08
+    continuous_dividends = 0.04
+    exp = self.get_barrier_option_test_vals("pui")
+    strikes = exp[0]
+    barriers = exp[1]
+    expected_price = exp[2]
+    is_call_options = exp[3]
+    is_barrier_down = exp[4]
+    is_knock_out = exp[5]
+    volatilities = 0.25
+
+    """
+    3 -> cdi
+    2 -> pdi
+    1 -> cui
+    0 -> pui
+    7 -> cdo
+    6 -> pdo
+    5 -> cuo
+    4 -> puo
+    """
+    price = tff.black_scholes.barrier_price(
+        volatilities=volatilities, strikes=strikes,
+        expiries=expiries, spots=spots,
+        discount_rates=discount_rates,
+        continuous_dividends=continuous_dividends,
+        barriers=barriers, rebates=rebates,
+        is_barrier_down=is_barrier_down,
+        is_knock_out=is_knock_out,
+        is_call_options=is_call_options)
+    self.assertAllClose(price, expected_price, 10e-3)
+
+  def get_barrier_option_test_vals(self, param):
```

Comment:
You can use parametrized test instead. See [here](https://github.com/google/tf-quant-finance/blob/1239ce3a22b4a6545d1600835333e8c9984ade7d/tf_quant_finance/black_scholes/approximations/american_option_test.py#L30)

Reply:
Parameterized!